entgql/annotation.go

```diff
@@ -59,6 +60,29 @@ type (
 		Value string        `json:"value,omitempty"`
 		Kind  ast.ValueKind `json:"kind,omitempty"`
 	}
+
+	// SkipFlag is a bit flag for the skip annotation .
+	SkipFlag int
+)
+
+const (
+	// SkipFlagType will skip generate the entity/field in the query
+	SkipFlagType SkipFlag = 1 << iota
+	// SkipFlagFieldEnum will skip generate Enum from the enum field
+	SkipFlagFieldEnum
+	// SkipFlagInput will skip mutation input for the entity/field
+	SkipFlagInput
+	// SkipFlagWhere will skip input filter for the entity/field
+	SkipFlagWhere
+	// SkipFlagOrder will skip generate sort order for the entity
+	SkipFlagOrder
```

Comment:
```suggestion
	SkipNode SkipFlag = 1 << iota
	// SkipEnum will skip generate Enum from the enum field
	SkipEnumField
	// SkipMutationInput will skip mutation input for the entity/field
	SkipMutationInput
	// SkipWhereInput will skip input filter for the entity/field
	SkipWhereInput
	// SkipOrderField will skip generate sort order for the entity
	SkipOrderField
```

I'm thinking about how the API will look and I shorter feels cleaner to me. Another point, what does it mean to `SkipOrderField` or `SkipEnumField`? Skip generating only GraphQL or also in Ent? Do we need a flag for both options?

Reply:
`SkipNode`: I thought about this name before, because we need to use it for all fields. So `SkipNode` does not seem appropriate. Do we need to add `SkipField` as well? `SkipType` seems more suitable to me, as it comes with `SkipInput`


> Another point, what does it mean to SkipOrderField or SkipEnumField? Skip generating only GraphQL or also in Ent

It should skip generating both Go code and GQL schema.

Reply:
> `SkipType` seems more suitable to me, as it comes with `SkipInput`

Right, let's use `SkipType`. I wonder how's the usage will look like:

```go
func (T) Annotations() []schema.Annotation {
    return []schema.Annotation{
        // Default to skip All.
        entgql.Skip(),

        // Or, skip only type and input generation.
        entgql.Skip(entgql.SkipType|entgql.SkipInput),
    }
}

func (T) Fields() []ent.Field {
    return []ent.Field{
        field.Int("age").
            Annotations(
                // Skip all.
                entgql.Skip(),

                // Or, skip only from input?
                entgql.Skip(entgql.SkipInput),
            ),
    }
}
```

This is the API you thought about or you have something else in mind?

Reply:
Yes, it's almost the same.

```go
func (T) Annotations() []schema.Annotation {
    return []schema.Annotation{
        // Default to skip All.
        entgql.Skip(),

        // Or, skip only type and input generation.
        entgql.Skip(entgql.SkipType, entgql.SkipMutationInput),
    }
}

func (T) Fields() []ent.Field {
    return []ent.Field{
        field.Int("age").
            Annotations(
                // Skip all.
                entgql.Skip(),

                // Or, skip only from input?
                entgql.Skip(entgql.SkipMutationInput),
            ),
    }
}
```